server/core/src/main/java/org/infinispan/server/core/configuration/ProtocolServerConfiguration.java

```diff
@@ -33,14 +33,16 @@ public abstract class ProtocolServerConfiguration<T extends ProtocolServerConfig
    public static final AttributeDefinition<Boolean> ZERO_CAPACITY_NODE = AttributeDefinition.builder(Attribute.ZERO_CAPACITY_NODE, false).immutable().build();
    public static final AttributeDefinition<String> SOCKET_BINDING = AttributeDefinition.builder(Attribute.SOCKET_BINDING, null, String.class).immutable().build();
    public static final AttributeDefinition<Boolean> IMPLICIT_CONNECTOR = AttributeDefinition.builder("implicit-connector", false).immutable().autoPersist(false).build();
+   public static final AttributeDefinition<Integer> MAX_BYTE_ARRAY_SIZE = AttributeDefinition.builder(Attribute.MAX_BYTE_ARRAY_SIZE, 10 * 1024 * 1024).immutable().build();
```

Comment:
It should be `ByteQuantity` instead of `Integer`, to use the units in the XML file. 

https://github.com/infinispan/infinispan/blob/fc71f27d2b6f8fd251e14d918b0520e7464af708/commons/all/src/main/java/org/infinispan/commons/util/ByteQuantity.java

Reply:
The XML attribute name seems longer to me. How about `max-length` instead of `max-byte-array-size` or `max-content-length` so all connectors have the same name?

Reply:
I had debated making the name the same, however it is different than the max-length provided by the REST endpoint as that is the entire request length and these are limits within a single request. HotRod/RESP/Memcached don't have a notion to tell you the request size ahead of time to validate.

I am fine with changing the names to something else though if you have better suggestions.

Reply:
I see.
`max-length` is the suggestion I have. I'm not sure if it is a better name or not :)

Reply:
let's use the same for every protocol. Any differences can be highlighted in the docs